tools/idl_validator.cpp

```diff
@@ -19,7 +21,19 @@ int main(
         idl::Context context;
         context.log_level(idl::log::LogLevel::xDEBUG);
         context.print_log(true);
-        context.include_paths.push_back("/opt/ros/foxy/share/");
+
+        // Introduce current ros2 paths
+        std::string distro(std::getenv("ROS_DISTRO"));
```

Comment:
If the environment variable is undefined, `std::getenv` returns a null pointer. The `std::string` constructor throws a `std::logic_error` instead of the graceful exit message. Suggest ` const char* distro = std::getenv("ROS_DISTRO");`, check for null pointer and construct distro string in else `.. + std::string(distro) +..`.

Is the `xtypes_idl_validator` only intended for ROS? The current implementation demands `ROS_DISTRO` to be defined.

Reply:
Got me! I turns out that construct a `std::string` from `nullptr` is undefined behavior. On MSVC leads to an empty string and in gcc a `std::logic_error` is raised. Crossplatform pitfalls 🙄.
We only used `xtypes_idl_validator` in an Integration-Service ROS2-SH related project. It is interesting for us to fail without a ROS2 overlay. But is a good hint too. Let's turn it into a warning.